File: modules/graph_init.py
import os
from py2neo import Graph
import logging

from modules import new_user
from modules import login


def connect(log):

    uri = "bolt://localhost:7688"
    user = "neo4j"
    password = os.getenv('BrainDBPassword')
    try:
        graph = Graph(uri, auth=(user, password))
    except Exception as e:
        log.info("No Database Found\n")

        log.info(e)

        log.info("Start Database and press press any key to continue\n")
        input()

    return graph

def create_system(log, graph):

    graph.run(f"\
              CREATE (m: Main),\
              (b: Brain),\
              (MM: MessageMaster),\
              (b)-[l: link]->(m),\
              (MM)-[n: link]->(b)\
              SET m.name = 'Main',\
              b.name = 'Brain',\
              MM.name = 'MessageMaster'\
              ")

    log.info("You must create a user profile to continue")

    new_user.user(log, graph)


def system_init(log, graph):

    main = graph.run(f"MATCH (n: Main)\
                     WHERE n.name='Main'\
                     RETURN (n)\
                     ").evaluate()

    if main == None:
        log.info("No data found. Initializing system.")
        create_system(log, graph)
    else:
        pass


def vault_init(graph):
    graph.run(f"MATCH (V: VAult) ")


def main(log):

    graph = connect(log)

    system_init(log, graph)

    return graph

File: Cephalopod_modules/Cephalopod_Brain/main/graph_init.js
const neo4j = require('neo4j-driver')

import { new_user } from "./new_user.js";

export default class graph_init {
    constructor() {
        uri = "bolt://localhost:7688";
        neo4j_user = "neo4j"
        neo4j_password = process.env.BrainDBPassword;
    
        const driver = neo4j.driver(uri, neo4j.auth.basic(neo4j_user, neo4j_password));
        const graph = driver.session();

        system_init(graph);

        return graph
    }
    
    system_init(graph){
        let main = graph.run(`MATCH (n: Main) WHERE n.name='Main' RETURN (n)`).evaluate();

        if (main == None) {
            console.log("No data found. Initializing system.");
            create_system(graph);
        }  
    }

    create_system(graph) {
        graph.run(`CREATE (m: Main), (b: Brain), (MM: MessageMaster), (b)-[l: link]->(m), (MM)-[n: link]->(b) SET m.name = 'Main', b.name = 'Brain', MM.name = 'MessageMaster' `);
        console.log("You must create a user profile to continue");
        new_user(graph);
    }
};